Engine/EditorUI/Panel/InspectorSubpanel/PanelComponent.cpp

```diff
@@ -1262,25 +1262,44 @@ void PanelComponent::ShowComponentAudioSourceWindow(ComponentAudioSource* compon
 		{
 			component_audio_source->SetSoundBank(selected_resource);
 			component_audio_source->modified_by_user = true;
+			component_audio_source->selected_event = "";
 		}
 		selected_resource = ImGui::ResourceDropper<SoundBank>();
 		if (selected_resource != 0)
 		{
 			component_audio_source->SetSoundBank(selected_resource);
 			component_audio_source->modified_by_user = true;
+			component_audio_source->selected_event = "";
 		}
 		if (component_audio_source->soundbank)
 		{
-			//static std::string soundbank;
-			ImGui::InputText("Awake Event Name ", &component_audio_source->awake_event);
+			std::string event_name = component_audio_source->selected_event != "" ? component_audio_source->selected_event : "-";
+			if (ImGui::BeginCombo("Event Name", event_name.c_str()))
+			{
+				for (auto& event_name : component_audio_source->soundbank->events)
+				{
+					if (ImGui::Selectable(event_name.c_str()))
+					{
+						component_audio_source->selected_event = event_name;
+						component_audio_source->modified_by_user = true;
+					}
+				}
+				ImGui::Separator();
+				ImGui::EndCombo();
+			}
+
 			if (ImGui::Button("Play Event"))
 			{
-				component_audio_source->PlayEvent(component_audio_source->awake_event);
+				if (event_name != "-")
+				{
+					component_audio_source->PlayEvent(component_audio_source->selected_event);
+				}
```

Comment:
This check should be inside the `ComponentAudioSource`. Instead of comparing `event_name`, comapre selected event.